src/sidebar/services/test/annotations-test.js

```diff
@@ -0,0 +1,254 @@
+'use strict';
+
+const annotations = require('../annotations');
+const EventEmitter = require('tiny-emitter');
+
+let searchClients;
+let longRunningSearchClient = false;
+class FakeSearchClient extends EventEmitter {
+  constructor(searchFn, opts) {
+    super();
+
+    assert.ok(searchFn);
+    searchClients.push(this);
+    this.cancel = sinon.stub();
+    this.incremental = !!opts.incremental;
+
+    this.get = sinon.spy(query => {
+      assert.ok(query.uri);
+
+      for (let i = 0; i < query.uri.length; i++) {
+        const uri = query.uri[i];
+        this.emit('results', [{ id: uri + '123', group: '__world__' }]);
+        this.emit('results', [{ id: uri + '456', group: 'private-group' }]);
+      }
+      if (!longRunningSearchClient) {
+        this.emit('end');
+      }
+    });
+  }
+}
+
+describe('annotations', () => {
+  let fakeStore;
+  let fakeApi;
+  let fakeAnnotationMapper;
+  let fakeStreamer;
+  let fakeStreamFilter;
+
+  let fakeUris;
+  let fakeGroupId;
+
+  beforeEach(() => {
+    searchClients = [];
```

Comment:
I sorta like this here too, feels like it resets a global var rather than having any one test set this and unset it again.
`longRunningSearchClient = false`

Reply:
Ah yes that would be good to have there!